File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/personal/PersonalNewAccountTest.java
package opensdk.sdk.apis.personal;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.PersonalNewAccountResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Personal RPC Test")
public class PersonalNewAccountTest {

    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC personal_newAccount")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        PersonalNewAccountResponse response = sdk.personal.newAccount("helloWorld").send();
        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/personal/PersonalNewAccount.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.PersonalApi(new OpenSdk.ApiClient(RPC));

describe('personal_newAccount API', () => {
    test('should return personal_newAccount', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            expect(data).toMatch(/^0x.*$/gm)
            done();
        };
        
        const passphrase = "helloWorld"

        sdk.newAccount({passphrase}, callbackOne);
    });
});

